src/test/groovy/graphql/execution/SubscriptionExecutionStrategyTest.groovy

```diff
@@ -335,4 +351,85 @@ class SubscriptionExecutionStrategyTest extends Specification {
+                    GraphQLError error = null
+                    if (index == 1) {
+                        error = mkError("1 is the loneliest number that you'll ever know")
+                    }
+                    // wrap inner result in DataFetcherResult
+                    def resultBuilder = DataFetcherResult.newResult().data(message).localContext(index)
+                    if (error != null) {
+                        resultBuilder.error(error)
+                    }
+                    return resultBuilder.build()
+                }
+                def publisher = new ReactiveStreamsObjectPublisher(10, objectMaker)
+                // we also use DFR here to wrap the publisher to show it can work
+                return DataFetcherResult.newResult().data(publisher).error(mkError("The top level field publisher can have errors")).build()
+            }
+
+        }
+
+        DataFetcher senderDF = new DataFetcher() {
+            @Override
+            Object get(DataFetchingEnvironment environment) throws Exception {
+                Message msg = environment.getSource()
+                if (msg.sender == "sender1") {
+                    return DataFetcherResult.newResult().data(msg.sender).error(mkError("Sub level fields can have errors")).build()
+                }
+                return msg.sender
+            }
+        }
+
+        GraphQL graphQL = buildSubscriptionQL(newMessageDF, senderDF, PropertyDataFetcher.fetching("text"))
+
+        def executionInput = ExecutionInput.newExecutionInput().query("""
+            subscription NewMessages {
+              newMessage(roomId: 123) {
+                sender
+                text
+              }
+            }
+        """).build()
+
+        when:
+
+        def executionResult = graphQL.execute(executionInput)
+
+        Publisher<ExecutionResult> msgStream = executionResult.getData()
+
+        def capturingSubscriber = new CapturingSubscriber<ExecutionResult>()
+        msgStream.subscribe(capturingSubscriber)
+
+        then:
+        Awaitility.await().untilTrue(capturingSubscriber.isDone())
+
+        executionResult.errors.size() == 1
+        executionResult.errors[0].message == "The top level field publisher can have errors"
+
+        def messages = capturingSubscriber.events
+        messages.size() == 10
+        for (int i = 0; i < messages.size(); i++) {
+            def message = messages[i]
+            // error handling on publisher events
+            if (i == 1) {
+                assert message.errors[0].message == "1 is the loneliest number that you'll ever know"
+                assert message.errors[1].message == "Sub level fields can have errors"
+            } else {
+                assert message.errors.isEmpty(), "There should be no errors present"
+            }
+            assert message.data == ["newMessage": [sender: "sender" + i, text: "text" + i]]
+        }
+    }
```

Comment:
A test of DataFectherResult + errors at the Publisher level and also at each streamed event level and also errors on sub levels